static/sass/_patterns_media-object--snap.scss

```diff
@@ -1,109 +1,107 @@
 // Based on the matrix component https://github.com/vanilla-framework/vanilla-framework/blob/develop/scss/_patterns_matrix.scss
 @mixin snapcraft-p-media-object--snap {
-  $matrix-img-width: map-get($icon-sizes, thumb--small);
-  $matrix-img-gutter: $sph-inter;
-
-  .p-media-object--snap {
-    display: block;
-    margin-bottom: $spv-inter--regular * 2;
-
-    &__img {
-      align-self: flex-start;
-      height: $matrix-img-width;
-      margin-bottom: $spv-inter--condensed;
-      margin-right: $matrix-img-gutter;
-      max-height: $matrix-img-width;
-      max-width: $matrix-img-width;
-      width: auto;
-    }
-
-    &__content {
-      display: flex;
-      flex: 1 1 auto;
-      flex-direction: column;
-      padding-right: $sph-intra;
+  %media-object {
+    .p-media-object__details {
+      overflow: hidden;
     }
 
-    &__title {
-      @extend %vf-heading-5;
-      margin-top: -$sp-unit;
+    .p-media-object__title {
+      max-width: 100%;
       overflow: hidden;
       text-overflow: ellipsis;
       white-space: nowrap;
-
-      &:hover {
-        text-decoration: underline;
-      }
     }
 
-    &__desc {
-      color: $color-mid-dark;
-      margin-bottom: $spv-nudge-compensation;
-      margin-top: - $sp-unit * 3;
-
-      > p:last-child {
-        margin-bottom: 0;
-      }
+    .p-media-object__content {
+      margin-top: -.5rem;
 
-      & + & {
-        margin-top: 0; // Negate nudge compensation for multiple .p-matrix__desc
+      p {
+        color: $color-mid-dark;
+        display: -webkit-box;
+        text-decoration: none;
+        overflow: hidden;
+        -webkit-line-clamp: 2;
+        -webkit-box-orient: vertical;
       }
 
-      @media (max-width: $breakpoint-heading-threshold) {
-        margin-top: - $sp-unit;
+      p + p {
+        color: $color-dark;
+        display: -webkit-box;
+        overflow: hidden;
+        -webkit-line-clamp: 2;
+        -webkit-box-orient: vertical;
       }
     }
 
-    &__publisher {
-      color: $color-mid-dark;
-      margin-bottom: $spv-nudge-compensation;
-      margin-top: - $sp-unit * 3;
-
-      > p:last-child {
-        margin-bottom: 0;
-      }
-
-      & + & {
-        margin-top: 0; // Negate nudge compensation for multiple .p-matrix__desc
-      }
+    &:hover {
+      text-decoration: none;
 
-      @media (max-width: $breakpoint-heading-threshold) {
-        margin-top: - $sp-unit;
+      .p-media-object__title {
+        text-decoration: underline;
       }
     }
+  }
 
-    &:hover {
-      text-decoration: none;
+  .p-media-object--snap {
+    @extend %media-object;
+
+    .p-media-object__image {
+      max-height: 3rem;
+      max-width: 3rem;
     }
 
-    &.is-placeholder {
-      .p-media-object--snap__img {
-        background: $color-light;
+    &.is--placeholder {
```

Comment:
Why double dashes? I think Vanilla uses one dash in their `is-something` class names?